homeassistant/components/mqtt_statestream/__init__.py

```diff
@@ -81,5 +78,28 @@ async def _state_publisher(entity_id, old_state, new_state):
                 encoded_val = json.dumps(val, cls=JSONEncoder)
                 await mqtt.async_publish(hass, mybase + key, encoded_val, 1, True)
 
-    async_track_state_change(hass, MATCH_ALL, _state_publisher)
+    @callback
+    def _ha_started(hass: HomeAssistant) -> None:
+        @callback
+        def _event_filter(evt: Event) -> bool:
+            entity_id: str = evt.data["entity_id"]
+            new_state: State | None = evt.data["new_state"]
+            if new_state is None:
+                return False
```

Comment:
I added a test for this, but somehow this seems not to be needed, or am I wrong here?

Reply:
Improved the tests